src/main/java/org/broadinstitute/hellbender/engine/LazyReadShard.java

```diff
@@ -26,7 +28,7 @@
  *
  * The reads in the shard can be filtered via {@link #setReadFilter} (no filtering is performed by default).
  */
-public final class ReadShard implements Iterable<GATKRead>, Locatable {
+public final class LazyReadShard implements Shard<GATKRead> {
```

Comment:
I think `LocalReadShard` is a better name (to contrast with `SparkReadShard`, and make it clear that this version is not meant for Spark use).